File: test-plugins/org.yakindu.sct.generator.c.test/gtests/EntryReactionActionTest/EntryReactionActionTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "EntryReactionAction.h"

EntryReactionAction handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		entryReactionAction_init(&handle);
	}
};

void init(){
	entryReactionAction_enter(&handle);
	entryReactionActionIface_raise_b(&handle);
	entryReactionAction_runCycle(&handle);
	entryReactionActionIface_raise_d(&handle);
	entryReactionAction_runCycle(&handle);
	entryReactionActionIface_set_enteredR1(&handle,false);
	entryReactionActionIface_set_enteredR2(&handle,false);
	entryReactionActionIface_set_enteredBdefault(&handle,false);
	entryReactionActionIface_set_enteredBother(&handle,false);
}

TEST_F(StatemachineTest, entryTransitionActionOnStatechartEnter) {					
	entryReactionAction_enter(&handle);
	EXPECT_TRUE(entryReactionActionIface_get_enteredR1(&handle));
	EXPECT_TRUE(entryReactionActionIface_get_enteredR2(&handle));
	EXPECT_TRUE(entryReactionActionIface_get_enteredBdefault(&handle));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBother(&handle));
}
TEST_F(StatemachineTest, entryOnRTS) {					
	init();
	entryReactionActionIface_raise_b(&handle);
	entryReactionAction_runCycle(&handle);
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR1(&handle));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR2(&handle));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBdefault(&handle));
	EXPECT_TRUE(entryReactionActionIface_get_enteredBother(&handle));
}
TEST_F(StatemachineTest, noEntryTransitionActionOnHistory) {					
	init();
	entryReactionActionIface_raise_d(&handle);
	entryReactionAction_runCycle(&handle);
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR1(&handle));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredR2(&handle));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBdefault(&handle));
	EXPECT_TRUE(!entryReactionActionIface_get_enteredBother(&handle));
}

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/EntryReactionActionTest.java
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/

package org.yakindu.sct.generator.java.test;

import org.junit.*;
import static org.junit.Assert.*;
import org.yakindu.scr.entryreactionaction.EntryReactionActionStatemachine;
import org.yakindu.scr.entryreactionaction.EntryReactionActionStatemachine.State;

/**
 * Unit TestCase for EntryReactionAction
 */
@SuppressWarnings("all")
public class EntryReactionActionTest {
	private EntryReactionActionStatemachine statemachine;	
	
	@Before
	public void setUp() {
		statemachine = new EntryReactionActionStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}
	
	@Test
	public void testEntryTransitionActionOnStatechartEnter() {
		statemachine.enter();
		assertTrue(statemachine.getEnteredR1());
		assertTrue(statemachine.getEnteredR2());
		assertTrue(statemachine.getEnteredBdefault());
		assertTrue(!statemachine.getEnteredBother());
	}
	@Test
	public void testEntryOnRTS() {
		init();
		statemachine.raiseB();
		statemachine.runCycle();
		assertTrue(!statemachine.getEnteredR1());
		assertTrue(!statemachine.getEnteredR2());
		assertTrue(!statemachine.getEnteredBdefault());
		assertTrue(statemachine.getEnteredBother());
	}
	@Test
	public void testNoEntryTransitionActionOnHistory() {
		init();
		statemachine.raiseD();
		statemachine.runCycle();
		assertTrue(!statemachine.getEnteredR1());
		assertTrue(!statemachine.getEnteredR2());
		assertTrue(!statemachine.getEnteredBdefault());
		assertTrue(!statemachine.getEnteredBother());
	}
	public void init() {
		statemachine.enter();
		statemachine.raiseB();
		statemachine.runCycle();
		statemachine.raiseD();
		statemachine.runCycle();
		statemachine.setEnteredR1(false);
		statemachine.setEnteredR2(false);
		statemachine.setEnteredBdefault(false);
		statemachine.setEnteredBother(false);
	}
}
